driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SyncMongoClient.java

```diff
@@ -34,6 +34,35 @@
 import static java.util.Objects.requireNonNull;
 
 public class SyncMongoClient implements MongoClient {
+
+    // Unfortunately this is the only way to wait for a query to be initiated, since Reactive Streams is asynchronous
+    // and we have no way of knowing. Tests which require cursor initiation to complete before execution of the next operation
+    // can set this to a positive value.  A value of 256 ms has been shown to work well.
+    // Defaults to 0
+    private static long sleepAfterCursorOpenMS;
+
+    // Unfortunately this is the only way to wait for close to complete, since it's asynchronous.
+    // This is inherently racy but there are not any other good options. Tests which require cursor cancellation to complete before
+    // execution of the next operation can set this to a positive value.  A value of 256 ms has been shown to work well.
+    // Defaults to 0
+    private static long sleepAfterCursorCloseMS;
+
+    public static void setSleepAfterCursorOpen(final long sleepMS) {
+        sleepAfterCursorOpenMS = sleepMS;
+    }
+
+    public static void setSleepAfterCursorClose(final long sleepMS) {
```

Comment:
It seems strange to have the sleep code itself in SyncMongoClient, rather than in the class that uses it.  Instead, I de-duplicated the wrapping by adding a private method in SyncMongoCursor.

I also don't need the 0 check for millis, so I removed it.